### PR TITLE
fix pool3d bug, test=develop

### PR DESCRIPTION
**Pull request by LDOUBLEV**
### PR types

Bug fixes
### PR changes

OPs
### Describe

fix pool3d bug, http://newicafe.baidu.com/v5/issue/DLTP-12154/show?cid=5

### CODE REVIEW

File: paddle/fluid/operators/math/pooling.cc [hunk 1]
```diff
@@ -712,10 +720,9 @@ class Pool3dFunctor<platform::CPUDeviceContext, PoolProcess, T> {
                   }
                 }
               }
-              int pool_size =
-                  (exclusive || adaptive)
-                      ? (dend - dstart) * (hend - hstart) * (wend - wstart)
-                      : ksize_depth * ksize_height * ksize_width;
+              if (exclusive || adaptive) {
```
Comment (willthefrog) [inline paddle/fluid/operators/math/pooling.cc:760]: this check seems redundant now, isn't `pool_size` always the same value?

**Review by willthefrog — CHANGES_REQUESTED** [review]
also fix the 2d and cuda version?

File: paddle/fluid/operators/math/pooling.cc (hunk 1 above)
Comment (LDOUBLEV) [reply paddle/fluid/operators/math/pooling.cc:760]: It is not redundant to update`pool_size`  when `padding ` is not zero.
The variable `pool_size` will be updated after running the following code.
```
pool_size = (dend - dstart) * (hend - hstart) * (wend - wstart);
dstart = std::max(dstart, 0);
hstart = std::max(hstart, 0);
wstart = std::max(wstart, 0);
dend = std::min(dend, input_depth);
hend = std::min(hend, input_height);
wend = std::min(wend, input_width);
```

**Comment by LDOUBLEV** [comment]
> also fix the 2d and cuda version?

Got it!

**Review by ceci3 — APPROVED** [review]

**Review by willthefrog — APPROVED** [review]
LGTM.

the cuda version seems unused, maybe remove later.

**Pull request merged**